source/opt/null_pass.h

```diff
@@ -35,6 +35,13 @@ namespace opt {
 
 // A null pass that does nothing.
 class NullPass : public Pass {
+ public:
+  NullPass() : Pass() {}
+  // Interfaces for different arguments.
+  NullPass(uint32_t arg) : Pass() { (void)arg; }
```

Comment:
I think you can get away with compiler warnings by omitting the parameter names.